apis/v1alpha1/baselineadminnetworkpolicy_types.go

```diff
@@ -0,0 +1,156 @@
+type BaselineAdminNetworkPolicyIngressRule struct {
+	// Name is an identifier for this rule, that may be no more than 100 characters
+	// in length. This field should be used by the implementation to help
+	// improve observability, readability and error-reporting for any applied
+	// BaselineAdminNetworkPolicies.
+	// +optional
+	// +kubebuilder:validation:MaxLength=100
+	Name string `json:"name,omitempty"`
+
+	// Action specifies the affect this rule will have on matching traffic,
+	// currently the following actions are supported:
+	// Allow: allows the selected traffic
+	// Deny: denies the selected traffic
+	// This field is mandatory.
+	Action BaselineAdminNetworkPolicyRuleAction `json:"action"`
+
+	// Ports allows for matching traffic based on port and protocols.
+	// If Ports is not set then traffic is not filtered via port.
+	// +optional
+	Ports []AdminNetworkPolicyPort `json:"ports,omitempty"`
+
+	// List of sources whose traffic this AdminNetworkPolicyRule applies to.
+	// If any adminNetworkPolicyPeer matches the source of incoming
+	// traffic then the specified action is applied.
+	// This field must be defined and contain at least one item.
+	// +kubebuilder:validation:MinItems=1
+	From []AdminNetworkPolicyPeer `json:"from"`
+}
+
+// AdminNetworkPolicyEgressRule describes an action to take on a particular
+// set of traffic originating from pods selected by a BaselineAdminNetworkPolicy's
+// Subject field. The traffic must match both ports and to.
+type BaselineAdminNetworkPolicyEgressRule struct {
+	// Name is an identifier for this rule, that may be no more than 100 characters
+	// in length. This field should be used by the implementation to help
+	// improve observability, readability and error-reporting for any applied
+	// BaselineAdminNetworkPolicies.
+	// +optional
+	// +kubebuilder:validation:MaxLength=100
+	Name string `json:"name,omitempty"`
+
+	// Action specifies the affect this rule will have on matching traffic,
+	// currently the following actions are supported:
+	// Allow: allows the selected traffic
+	// Deny: denies the selected traffic
+	// This field is mandatory.
+	Action BaselineAdminNetworkPolicyRuleAction `json:"action"`
+
+	// Ports allows for matching traffic based on port and protocols.
+	// If Ports is not set then traffic is not filtered via port.
+	// +optional
+	Ports []AdminNetworkPolicyPort `json:"ports,omitempty"`
+
+	// List of destinations whose traffic this adminNetworkPolicyRule applies to.
+	// If any adminNetworkPolicyPeer matches the destination of outgoing
+	// traffic then the specified action is applied.
+	// This field must be defined and contain at least one item.
+	// +kubebuilder:validation:MinItems=1
+	To []AdminNetworkPolicyPeer `json:"to"`
+}
+
+// BaselineAdminNetworkPolicyRuleAction string describes the BaselineAdminNetworkPolicy
+// action type.
+// +enum
+type BaselineAdminNetworkPolicyRuleAction string
+
+const (
+
```

Comment:
excess blank line... do we not have a `make verify` that runs gofmt? Should get that at some point

Reply:
We do run go-fmt whenever we generate the CRDs... Not sure why it didn't pick this up